src/containers/my-cooperations/cooperation-activities-list/CooperationActivitiesList.tsx

```diff
@@ -22,23 +38,27 @@ const CooperationActivitiesList = () => {
     isNewActivity,
     sections
   } = useAppSelector(cooperationsSelector)
+
   const dispatch = useAppDispatch()
   const Id = uuidv4()
 
+  // This logic looks very complicated and seems that it doesn't work
+  // Why do we need to store some flags for user actions?
+  // isAddedClicked works even when we don't click, that adds unnecessary sections
   useEffect(() => {
     if (!sections?.length && !isAddedClicked && isNewActivity) {
       addNewSection()
     }
 
-    if (selectedCourse && !sections.length && isAddedClicked) {
+    if (selectedCourse && !sections?.length && isAddedClicked) {
```

Comment:
This was needed to fix undefined issues after resolving conflicts with the PR: https://github.com/ita-social-projects/SpaceToStudy-Client/pull/2053. This is not working as expected in this PR and needs further investigation of all the useEffect logic.